### ISSUE TITLE
[Project 2] Question on using mutex

### ISSUE THREAD

**Issue by HunHuh** [swsnu/osspr2021#52]
Hi again (sorry for asking too much questions...haha...)
This question is on using mutex locks.

When Thread2 calls mutex lock when Thread1 has already called the mutex lock, Thread2 would be blocked.
In this state, if the following action of Thread1 is to unlock mutex and then lock again right after the unlock,
is it guaranteed that Thread2 will be unblocked, 
or is it possible that Thread1 acquires the lock again and Thread2 will still be blocked?
In other words, is it guaranteed that the one of the blocked threads acquires the lock when mutex unlock is called?

Thank you.

**Comment by jaywonchung** [comment]
No, you don't have to be sorry! The questions you ask will benefit others, too.

I talked with the other TAs, and we decided that this is an issue that you should figure out yourself. First, review the *definition* of mutex. Then, read the linux source code that implements `mutex`. The post below may help.

Post:
- http://jake.dothome.co.kr/mutex/

**Comment by HunHuh** [comment]
Sure! Thank you for your help!
